commands/generate.js

```diff
@@ -0,0 +1,111 @@
+'use babel';
+
+var fs = require('fs');
+
+var getProjectRoot = function getProjectRoot() {
+	'use strict';
+
+	return atom.project.getPaths()[0];
+};
+
+var getConfig = function getConfig() {
+	'use strict';
+
+	return new Promise (function(resolve) {
+		var config = {
+			core: atom.config.get('core'),
+			editor: atom.config.get('editor'),
+			whitespace: atom.config.get('whitespace')
+		};
+
+		var data = {
+			root: true,
+			end_of_line: process.platform === 'win32' ? 'crlf' : 'lf',
+			charset: config.core.fileEncoding || 'utf-8',
+			insert_final_newline: config.whitespace.ensureSingleTrailingNewline,
+			trim_trailing_whitespace: config.whitespace.removeTrailingWhitespace
+		};
+
+		if (config.editor.softTabs) {
+			data.indent_style = 'space';
+			data.indent_size = config.editor.tabLength;
+		} else {
+			data.indent_style = 'tab';
+			data.tab_width = config.editor.tabLength;
+		}
+
+		resolve(data);
+	});
+};
+
+var getOutput = function getOutput(data) {
+	'use strict';
+
+	return new Promise(function(resolve) {
+		var output = '# EditorConfig is awesome: http://EditorConfig.org\n';
+		output += '# top-most EditorConfig file\n';
+		output += 'root = ' + data.root+ '\n\n';
+		output += '[*]\n\n';
+		output += 'charset = ' + data.charset+ '\n';
+		output += 'end_of_line = ' + data.end_of_line+ '\n';
+		output += 'insert_final_newline = ' + data.insert_final_newline+ '\n';
+		output += 'trim_trailing_whitespace = ' + data.trim_trailing_whitespace + '\n';
+		output += 'indent_style = ' + data.indent_style+ '\n';
+		if (data.indent_style === 'space') {
+			output += 'indent_size = ' + data.indent_size+ '\n';
+		} else {
+			output += 'tab_width = ' + data.tab_width+ '\n';
+		}
+
+		resolve(output);
+	});
+};
+
+var saveConfigFile = function saveConfigFile(output) {
+	'use strict';
+
+	return new Promise (function(resolve, reject) {
+		var projectRoot = getProjectRoot();
+		var files = fs.readdirSync(projectRoot).filter(file => file === '.editorconfig');
```

Comment:
No point in reading the directory. Just try `fs.access` the `.editorconfig` file.